pallets/phala-world/src/lib.rs

```diff
@@ -0,0 +1,915 @@
+		OverlordNotSet,
+		RequireOverlordAccount,
+		InvalidStatusType,
+	}
+
+	// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+	// These functions materialize as "extrinsics", which are often compared to transactions.
+	// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		T: pallet_uniques::Config<ClassId = CollectionId, InstanceId = NftId>,
+	{
+		/// Claim a spirit for users that are on the whitelist. This whitelist will consist of a
+		/// a serial id and an account id that is signed by the admin account. When a user comes
+		/// to claim their spirit, they will provide a serial id & will be validated as an
+		/// authenticated claimer
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic.
+		/// - serial_id: The serial id of the spirit to be claimed.
+		/// - signature: The signature of the account that is claiming the spirit.
+		///   //Sr25519Signature
+		/// - metadata: The metadata of the account that is claiming the spirit.
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn claim_spirit(
+			origin: OriginFor<T>,
+			serial_id: SerialId,
+			signature: sr25519::Signature,
+			metadata: BoundedVec<u8, T::StringLimit>,
+		) -> DispatchResult {
+			ensure!(CanClaimSpirits::<T>::get(), Error::<T>::SpiritClaimNotAvailable);
+			let sender = ensure_signed(origin)?;
+			let overlord = Overlord::<T>::get().ok_or(Error::<T>::OverlordNotSet)?;
+			// Has the SerialId already been claimed
+			ensure!(
+				!ClaimedSpirits::<T>::contains_key(serial_id),
+				Error::<T>::SpiritAlreadyClaimed
+			);
+			// Check if valid SerialId to claim a spirit
+			ensure!(
+				Self::verify_claim(sender.clone(), metadata.clone(), signature),
+				Error::<T>::ClaimVerificationFailed
+			);
+			// Mint new Spirit and transfer to sender
+			pallet_rmrk_core::Pallet::<T>::mint_nft(
+				Origin::<T>::Signed(overlord).into(),
+				sender.clone(),
+				SPIRIT_COLLECTION_ID,
+				None,
+				None,
+				metadata,
+			)?;
+			ClaimedSpirits::<T>::insert(serial_id, true);
+
+			Self::deposit_event(Event::SpiritClaimed { serial_id, owner: sender });
+
+			Ok(())
+		}
+
+		/// Buy a rare egg of either type Legendary or Founder. Both Egg types will have a set
+		/// price. These will also be limited in quantity and on a first come, first serve basis.
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic.
+		/// - egg_type: The type of egg to be purchased.
+		/// - race: The race of the egg chosen by the user.
+		/// - career: The career of the egg chosen by the user or auto-generated based on metadata
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn buy_rare_egg(
+			origin: OriginFor<T>,
+			egg_type: EggType,
+			race: RaceType,
+			career: CareerType,
+			metadata: BoundedVec<u8, T::StringLimit>,
```

Comment:
I'll add this to next PR. I have 2 ways to resolve this by using `pallet_uniques` functions:
- `set_metadata`
- `set_attribute`
I lean towards `set_attribute` since this is a `key` `value` and we can access the `Attribute` storage and makes it easier to retrieve this info later on.